src/software/sensor_fusion/filter/ball_filter.h

```diff
@@ -52,25 +45,67 @@ class BallFilter
 
     /**
      * Creates a new Ball Filter
-     *
-     * @param min_buffer_size The minimum size of the buffer the filter will use to filter
-     * the ball. The buffer will shrink to this size as the ball speeds up
-     * @param min_buffer_size The maximum size of the buffer the filter will use to filter
-     * the ball. The buffer will grow to this size as the ball slows down
      */
-    explicit BallFilter(unsigned int min_buffer_size = DEFAULT_MIN_BUFFER_SIZE,
-                        unsigned int max_buffer_size = DEFAULT_MAX_BUFFER_SIZE);
+    explicit BallFilter();
 
     /**
      * Filters the new ball detection data, and returns the updated state of the ball
-     * given the new data *
-     * @param current_ball_state The current state of the Ball
+     * given the new data
+     *
      * @param new_ball_detections A list of new Ball detections
+     * @param filter_area The area within which the ball filter will work. Any detections
+     * outside of this area will be ignored.
      *
      * @return The updated state of the ball given the new data
      */
-    std::optional<TimestampedBallState> getFilteredData(
-        const std::vector<BallDetection> &new_ball_detections, const Field &field);
+    std::optional<TimestampedBallState> estimateBallState(
+        const std::vector<BallDetection>& new_ball_detections,
+        const Rectangle& filter_area);
+
+   private:
+    /**
+     * A simple struct we use to pass around velocity estimate data
+     */
+    struct BallVelocityEstimate
+    {
+        Vector average_velocity;
+        double average_velocity_magnitude;
+        // The average of the max velocity magnitude and min velocity magnitude
+        double min_max_magnitude_average;
+    };
+
+    /**
+     * A simple struct to pass around linear regression data
+     */
+    struct LinearRegressionResults
+    {
+        Line regression_line;
+        double regression_error;
+    };
+
+    /**
+     * Adds ball detections to the buffer stored by this filter. This function will ignore
+     * data that is outside of the filter_area, and data that is too far away from the
```

Comment:
instead of "and", is it better to use "since that data..."? From what I understood, the purpose of the filter_area is to determine data that is too far away.

Reply:
Yes `filter_area` is used for that, but we also check if a new detection is "possible" to be a ball. We can estimate the velocity the ball would have to travel at to reach the new detection position from its current filtered position. If that velocity is significantly above the legal ball speed limit, we can assume the new detection is noise and not a ball.

I've reworded the comment a bit to try make the conditions more clear.